tests/test_views.py

```diff
@@ -48,72 +59,357 @@ def test_shorten_acquisition_url(self):
+            "download/server/thank-you.html",
+            version="",
+            architecture="",
+            lts="",
+        )
+
+
+class TestAccountQuery(BaseViewTestCase):
+    """Tests for account_query without mocking user_info."""
+
+    def test_account_query_with_user(self):
+        import flask
+
+        with self.app.test_request_context("/any-url"):
+            flask.session["openid"] = {
+                "fullname": "Alice Example",
+                "email": "alice@example.com",
+                # omit flags so defaults (False) are exercised
+            }
+            flask.session["authentication_token"] = "tok123"
+            resp = account_query()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(
+            resp.get_json(),
+            {
+                "account": {
+                    "fullname": "Alice Example",
+                    "email": "alice@example.com",
+                    "authentication_token": "tok123",
+                    "is_community_member": False,
+                    "is_credentials_admin": False,
+                    "is_credentials_support": False,
+                }
+            },
+        )
+
+    def test_account_query_no_user(self):
+        with self.app.test_request_context("/any-url"):
+            resp = account_query()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.get_json(), {"account": None})
+
+class TestBuildingTutorialsQuery(BaseViewTestCase):
+    def _make_tutorials_docs(self, tutorials):
+        class DummyAPI:
+            base_url = "https://discourse.example.com"
+
+        class DummyParser:
+            def __init__(self, tutorials):
+                self.tutorials = tutorials
+                self.index_topic = 999
+                self.api = DummyAPI()
+
+            def parse(self):
+                pass
+
+            def parse_topic(self, topic):
+                pass
+
+        class DummyDocs:
+            def __init__(self, tutorials):
+                self.parser = DummyParser(tutorials)
+
+        return DummyDocs(tutorials)
+
+    def test_building_tutorials_query(self):
+        tutorials = [
+            {
+                "title": "Tutorial 1",
+                "slug": "tutorial-1",
+                "categories": ["c1", "c2"],
+                "author": "Author 1",
+                "difficulty": 3,
+                "created_at": "2023-01-01T00:00:00Z",
+            },
+            {
+                "title": "Tutorial 2",
+                "slug": "tutorial-2",
+                "categories": ["c2"],
+                "author": "Author 2",
+                "difficulty": 2,
+                "created_at": "2023-02-01T00:00:00Z",
+            },
+            {
+                "title": "Tutorial 3",
+                "slug": "tutorial-3",
+                "categories": ["c1"],
+                "author": "Author 3",
+                "difficulty": 1,
+                "created_at": "2023-03-01T00:00:00Z",
```

Comment:
We can also add a test for an unexpected tutorial field and check if the index builds as expected without the invalid entry
```suggestion
                "created_at": "2023-03-01T00:00:00Z",
                "unexpected_field": "test"
```